lib/integration_test.go

```diff
@@ -20,6 +20,8 @@ import (
 )
 
 func TestTwoActorRegistryIntegration(t *testing.T) {
+	t.Skip("TODO(dustmop): meaning of source changed, this needs to be fixed")
```

Comment:
This is an important test. Can't skip. let's instead figure out what's going on before merging

Reply:
The spike at https://github.com/qri-io/qri/commit/c5c65b84afca781587251da5cb21a04350b05c36 was super helpful. Pushed another commit that fixes the test.

What was happening is that the implementation of `dataset.pull` overrides the source with "network" if a blank source is passed in. However, this change is moving to a world where the `scope` locks-in the source, and implementations should not be allowed to change it.

I see 3 options going forward:

1) Require users of `pull` to always do this:
```
instance.WithSource("network").Pull(...)
```
This is the change I went with because it seems easiest. Now `cmd` calls WithSource('network'). In the future we can add a method to scope to get (but not set) the source, and pull can fail if it's not "network".

2) Add a method to scope to override the source. pull can call this to set the source. I'm not a fan of this option because it goes against what this PR is trying to do, but if it's only needed in a few places it's not that bad.

3) Add a field to Attributes to set the default source for commands. This will usually be empty, but can be "network" for pull. I don't think we should do this until we find more places that need it.

Reply:
Very much agree this is the right approach for now. This PR makes the need for the override explicit, which I'm a huge fan of. Option 3 (adding a field to a method set's returned Attributes) seems like a reasonable design choice, but I'm with you that it's too soon to make such a change